browser/ui/split_view/split_view_link_redirect_utils.cc

```diff
@@ -0,0 +1,86 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/ui/split_view/split_view_link_redirect_utils.h"
+
+#include "chrome/browser/ui/ui_features.h"
+#include "components/tabs/public/split_tab_collection.h"
+#include "components/tabs/public/split_tab_data.h"
+#include "components/tabs/public/tab_collection.h"
+#include "components/tabs/public/tab_interface.h"
+#include "content/public/browser/navigation_controller.h"
+#include "content/public/browser/web_contents.h"
+#include "content/public/common/referrer.h"
+#include "ui/base/page_transition_types.h"
+#include "url/gurl.h"
+
+namespace {
+
+content::WebContents* GetRightPaneIfLinked(content::WebContents* source) {
+  if (!source) {
+    return nullptr;
+  }
```

Comment:
I think this should never happen based on current usage. how about changing it to `CHECK(source)`